src/Analysis/Ast/Impl/Analyzer/AnalysisModuleKey.cs

```diff
@@ -28,15 +28,14 @@ namespace Microsoft.Python.Analysis.Analyzer {
         public bool IsTypeshed { get; }
         public bool IsNonUserAsDocument { get; }
 
-        public AnalysisModuleKey(IPythonModule module) {
-            Name = module.Name;
-            FilePath = module.ModuleType == ModuleType.CompiledBuiltin ? null : module.FilePath;
-            IsTypeshed = module is StubPythonModule stub && stub.IsTypeshed;
-            IsNonUserAsDocument = (module.IsNonUserFile() || module.IsCompiled()) && module is IDocument document && document.IsOpen;
-        }
+        public AnalysisModuleKey(IPythonModule module) : this(
+            module.Name,
+            module.ModuleType == ModuleType.CompiledBuiltin ? null : module.FilePath,
+            module is StubPythonModule && ((StubPythonModule)module).IsTypeshed,
+            (module.IsNonUserFile() || module.IsCompiled()) && module is IDocument && ((IDocument)module).IsOpen) { }
```

Comment:
As above

Reply:
You can check other files in the project, we prefer `module is IDocument document && document.IsOpen` rather than casts

Reply:
ya. I prefer that as well. let me do some trick so that I can do that instead of putting code directly in this(..) so I can use that construct.